pyogrio/tests/conftest.py

```diff
@@ -23,10 +23,18 @@ def data_dir():
     return _data_dir
 
 
-@pytest.fixture(scope="session")
-def naturalearth_lowres():
-    return _data_dir / Path("naturalearth_lowres/naturalearth_lowres.shp")
-
+@pytest.fixture(scope="function")
+def naturalearth_lowres(tmp_path, suffix: str = ".shp"): #request):
```

Comment:
I tested it while debugging and was sure it worked... but I just checked again and indeed it doesn't :-(. Weird.

Before I tried that, I also found that the parameters are also available in the request object... like this: 
```
ext = request._pyfuncitem.callspec.params.get("ext"))
```
Does that seem like a reasonable approach? The reason I look for a different approach than the fixture params is because those always "broadcast"/"cross join" with all `mark.parametrize` params to all combinations while this is often not needed/wanted.

Reply:
For the 
> Note that converting this particular dataset to GPKG causes GDAL to emit a warning to stdout (we don't capture it):"
> `geometry of type POLYGON is inserted into layer naturalearth_lowres of geometry type MULTIPOLYGON...`
Yes, I also just made a pull rquest for the promote_to_multi keyword, that should solve that.

Reply:
> But possibly for the pyogrio tests this isn't really an issue/worth using a private variable in pytest.

My preference would be to stay away from private variables in other packages.  I think if we have a specific fixture that gives us variants of extensions (compared to the default `naturalearth_lowres`), we can be careful about not using this more than expected - and it gives us the ability to leverage some of those cross-product combinations with other parameters where doing so may be helpful.

Reply:
If you want to be able to parametrize the fixture in the test itself and not in the fixture definition, you can do that using the `indirect` keyword (see https://stackoverflow.com/a/33879151), which avoids using the private pytest call (because then you can use the normal `request.param`).

Now, whether we want to directly or indirectly parametrize the fixture might depend on how we are going to use it (if we mostly parametrize it with the same parameters, it might make it more complicated to always have to do this in the test indirectly)

Reply:
I tried to combine the best of both worlds as such:

- The "current" naturalearth_lowres fixture can now optionally receive an "indirect" parameter to get another file type version of the naturalearth_lowres. If it is not provided, the shapefile will be returned
```
@pytest.mark.parametrize(
        "naturalearth_lowres, expected_ext",
        [(".gpkg", ".gpkg"), (None, ".shp")],
        indirect=["naturalearth_lowres"])
def test_fixture_naturalearth_lowres(naturalearth_lowres, expected_ext):
    # Test the fixture with "indirect" parameter
    assert naturalearth_lowres.suffix == expected_ext
    df = read_dataframe(naturalearth_lowres)
    assert len(df) == 177
```
- A naturalearth_lowres_all_ext fixture is available as well and this will broadcast to all supported extensions (except for .fgb, but this is solved in another pull request)

What do you think?

I already applied naturalearth_lowres_all_ext to some tests, but for some other tests this exposes underlying bugs... so I reverted those changes for the time being. I'll create bug issues for these cases?

Reply:
That's a nice solution!